### PR TITLE
Phase2-hgx94 Remove the sqaure cell option for HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
Clean it up a bit before inserting post-TDR geometry

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/CaloTopology/src/HGCalTopology.cc [hunk 1]
```diff
@@ -198,6 +198,7 @@ DetId HGCalTopology::encode(const HGCalTopology::DecodedDetId& id_) const {
 DetId HGCalTopology::changeXY(const DetId& id, int nrStepsX,
 			      int nrStepsY ) const {
 
+  /*
```
Comment (kpedro88) [inline Geometry/CaloTopology/src/HGCalTopology.cc]: why comment out this code instead of just deleting it?

Comment (bsunanda) [reply Geometry/CaloTopology/src/HGCalTopology.cc]: The newer geometry definition will use this interface later. I like to keep the interface - might delete the commented out code here.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Calo/src/HGCSD.cc [hunk 2]
```diff
@@ -148,43 +148,37 @@ uint32_t HGCSD::setDetUnitId(const G4Step * aStep) {
...
+			 << touch->GetVolume(2)->GetName() 
+			 << ":" << touch->GetReplicaNumber(2) << "   "
+			 << " layer:module:cell " << layer << ":" << module 
+			 << ":" << cell <<" Material " << mat->GetName()<<":"
+			 << aStep->GetPreStepPoint()->GetMaterial()->GetRadlen();
 #endif
-    if (aStep->GetPreStepPoint()->GetMaterial()->GetRadlen() > 100000.) return 0;
-  }
-
+  if (aStep->GetPreStepPoint()->GetMaterial()->GetRadlen() > 100000.) return 0;
```
Comment (civanch) [inline SimG4CMS/Calo/src/HGCSD.cc]: @bsunanda , why radiation length is checked here? What means very large value? I would guess it never happens.

File: SimG4CMS/Calo/src/HGCSD.cc [hunk 3]
```diff
@@ -148,43 +148,37 @@ uint32_t HGCSD::setDetUnitId(const G4Step * aStep) {
...
+  if (touch->GetHistoryDepth() == levelT_) {
     layer  = touch->GetReplicaNumber(0);
-    module = touch->GetReplicaNumber(1);
-  } else {
-    if (touch->GetHistoryDepth() == levelT_) {
-      layer  = touch->GetReplicaNumber(0);
-      module = -1;
-      cell   = -1;
+    module = -1;
+    cell   = -1;
```
Comment (civanch) [inline SimG4CMS/Calo/src/HGCSD.cc]: @bsunanda , why we need -1 here? initially they were initialized as 0.

File: SimG4CMS/Calo/src/HGCSD.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/Calo/src/HGCSD.cc]: This was done while doing the testbed studies. I don't remember the reason for this fix

File: SimG4CMS/Calo/src/HGCSD.cc (hunk 3 above)
Comment (bsunanda) [reply SimG4CMS/Calo/src/HGCSD.cc]: I shall in any case revise HGCSD later when I have the new version of HGCal geometry.  I want to remove only the square cell part right now to ease the changes needed for the future version

Comment (bsunanda) [reply SimG4CMS/Calo/src/HGCSD.cc]: There is some intricacies for partial layers - I shall see what can be done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda , for some reason WF 10224.0 demonstrate ~10000 differences in regression. It seems, that your recent commit should not have such effect but it is. For any case, I will restart the test.

**Comment by civanch** [comment]
please test

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h [hunk 4]
```diff
@@ -29,14 +29,8 @@ class HGCalDDDConstants {
 
   std::pair<int,int>  assignCell(float x, float y, int lay, int subSec,
 				 bool reco) const;
-  std::pair<int,int>  assignCellSquare(float x, float y, float h, float bl, 
-				       float tl, float alpha, 
-				       float cellSize) const;
```
Comment (ianna) [inline Geometry/HGCalCommonData/interface/HGCalDDDConstants.h]: @bsunanda - now that there is only hexagonal geometry left shouldn't "Hexagon" be dropped from the methods names?

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 5]
```diff
@@ -15,26 +15,8 @@ constexpr double k_ScaleFromDDD = 0.1;
...
-			      << "maximum of " << maxCells(false) << ":" 
-			      << maxCells(true) << " cells";
-#ifdef EDM_ML_DEBUG
-    std::cout << "HGCalDDDConstants initialized for " << name << " with " 
-	      << layers(false) << ":" << layers(true) << " layers, " 
-	      << sectors() << " sectors and maximum of " << maxCells(false)
-	      << ":" << maxCells(true) << " cells" << std::endl;
-#endif
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - there are only these two modes. It's better to use an assert.

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 6]
```diff
@@ -113,60 +95,14 @@ std::pair<int,int> HGCalDDDConstants::assignCell(float x, float y, int lay,
   std::pair<int,int> cellAssignment(-1,-1);
   int i = index.first;
   if (i < 0) return cellAssignment;
-  if (mode_ == HGCalGeometryMode::Square) {
-    float alpha, h, bl, tl;
-    getParameterSquare(i,subSec,reco,h,bl,tl,alpha);
-    cellAssignment = assignCellSquare(x, y, h, bl, tl, alpha, index.second);
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 7]
```diff
@@ -268,22 +161,17 @@ std::vector<HGCalParameters::hgtrform> HGCalDDDConstants::getTrForms() const {
...
-    cellmax = maxCells(lay,reco);
-    modmax  = sectors();
-    ok      = ((lay > 0 && lay <= (int)(layers(reco))) && 
-	       (mod > 0 && mod <= modmax) &&
-	       (cell >=0 && cell <= cellmax));
-  } else {
+  bool ok(false), okMod(false);
+  int  cellmax(0);
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 8]
```diff
@@ -347,22 +234,13 @@ unsigned int HGCalDDDConstants::layersInit(bool reco) const {
...
-    getParameterSquare(i,type,reco,h,bl,tl,alpha);
-    float cellSize = index.second;
-    x              = (kxy.first+0.5)*cellSize;
-    if      (alpha < 0) x -= 0.5*(tl+bl);
-    else if (alpha > 0) x -= 0.5*(tl+bl);
-    if (type != 1) x = -x;
-    y              = ((kxy.second+0.5)*cellSize-h);
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 9]
```diff
@@ -412,11 +290,8 @@ int HGCalDDDConstants::maxCells(int lay, bool reco) const {
   std::pair<int,float> index = getIndex(lay, reco);
   int i = index.first;
   if (i < 0) return 0;
-  if (mode_ == HGCalGeometryMode::Square) {
-    float h, bl, tl, alpha;
-    getParameterSquare(i,0,reco,h,bl,tl,alpha);
-    return maxCellsSquare(h, bl, tl, alpha, index.second);
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 10]
```diff
@@ -426,36 +301,19 @@ int HGCalDDDConstants::maxCells(int lay, bool reco) const {
...
   int kymax(0);
   std::pair<int,float> index = getIndex(lay, reco);
   int i = index.first;
   if (i < 0) return kymax;
-  if (mode_ == HGCalGeometryMode::Square) {
-    float h = (reco) ? hgpar_->moduleHR_[i] : hgpar_->moduleHS_[i];
-    kymax   = floor((2*h)/index.second);
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 11]
```diff
@@ -481,72 +339,14 @@ int HGCalDDDConstants::modulesInit(int lay, bool reco) const {
...
   int i = index.first;
   std::vector<int> ncell;
   if (i >= 0) {
-    if (mode_ == HGCalGeometryMode::Square) {
-      float h, bl, tl, alpha;
-      getParameterSquare(i,0,reco,h,bl,tl,alpha);
-      return numberCellsSquare(h, bl, tl, alpha, index.second);
-    } else {
+    if ((mode_ == HGCalGeometryMode::Hexagon) ||
+	(mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 12]
```diff
@@ -605,29 +392,8 @@ std::pair<int,int> HGCalDDDConstants::simToReco(int cell, int lay, int mod,
...
-      kx += floor(((iky+1)*cellSize+b+k_horizontalShift*cellSize)/(a*cellSize));
-#ifdef EDM_ML_DEBUG
-    std::cout << "simToReco: input " << cell << ":" << lay << ":" << half
-	      << " kxy " << kxy.first << ":" << kxy.second << " output "
-	      << kx << ":" << depth << " cell factor=" 
-	      << hgpar_->cellFactor_[i] << std::endl;
-#endif
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 13]
```diff
@@ -784,33 +550,14 @@ std::pair<int,float> HGCalDDDConstants::getIndex(int lay, bool reco) const {
   if (reco && lay>(int)(hgpar_->depthIndex_.size()))  return std::pair<int,float>(-1,0);
   int   indx(0);
   float cell(0);
-  if (mode_ == HGCalGeometryMode::Square) {
-    indx = (reco ? hgpar_->depthIndex_[lay-1] : hgpar_->layerIndex_[lay-1]);
-    cell = (reco ? hgpar_->moduleCellR_[indx] : hgpar_->moduleCellS_[indx]);
-  } else {
+  if ((mode_ == HGCalGeometryMode::Hexagon) ||
+      (mode_ == HGCalGeometryMode::HexagonFull)) {
```
Comment (ianna) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: @bsunanda - same here

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h (hunk 4 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalDDDConstants.h]: Yan - there will be new versions to be introduced soon. I would like to leave this as it is. New mode will be introduced next

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 5 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: There will be some more new modes which will come

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 7 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Please see above comments

**Comment by bsunanda** [comment]
@ianna This is an interim step  to get rid of one mode and some new mode will come next. I would like to have these "hexagon" stuff to be kept

**Comment by ianna** [comment]
+1

**Comment by bsunanda** [comment]
@civanch Can you please approve this?

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**